examples/google-ads-account-analyzer-demo/analyzer.py

```diff
@@ -0,0 +1,224 @@
+    customer_service = google_ads_client.get_service("CustomerService")
+    # A collection of customer IDs to handle.
+    seed_customer_ids = []
+
+    # If a manager ID was provided in the customer ID parameter, it will be
+    # the only ID in the list. Otherwise, we will issue a request for all
+    # customers accessible by this authenticated Google account.
+    if customer_id:
+        seed_customer_ids = [customer_id]
+    else:
+        print(
+            "No manager ID is specified. The example will print the "
+            "hierarchies of all accessible customer IDs."
+        )
+        # Starting v10 this will list all customers, not just customers with ENABLED status.
+        # If you wish to only show ENABLED customers, you can further filter by
+        # Customer.status field.
+        accessible_customers = customer_service.list_accessible_customers()
+        customer_resource_names = accessible_customers.resource_names
+        print(f"Total results: {len(customer_resource_names)}")
+        for resource_name in customer_resource_names:
+            print(f'Customer resource name: "{resource_name}"')
+            seed_customer_ids.append(resource_name.split("/")[1])
+
+    # Creates a query that retrieves all child accounts of the manager
+    # specified in search calls below.
+    query = """
+        SELECT
+            customer_client.client_customer,
+            customer_client.level,
+            customer_client.manager,
+            customer_client.descriptive_name,
+            customer_client.currency_code,
+            customer_client.time_zone,
+            customer_client.id
+        FROM customer_client
+        WHERE customer_client.level <= 1"""
+
+    for seed_customer_id in seed_customer_ids:
+        # Performs a breadth-first search to build a Dictionary that maps
+        # managers to their child accounts (customerIdsToChildAccounts).
+        unprocessed_customer_ids = [seed_customer_id]
+        customer_ids_to_child_accounts = dict()
+        root_customer_client = None
+
+        while unprocessed_customer_ids:
+            customer_id = unprocessed_customer_ids.pop(0)
+            response = googleads_service.search(
+                customer_id=customer_id, query=query
+            )
+
+            # Iterates over all rows in all pages to get all customer
+            # clients under the specified customer's hierarchy.
+            for row in response:
+                customer_client = row.customer_client
+
+                # The customer client that with level 0 is the specified
+                # customer.
+                if customer_client.level == 0:
+                    if root_customer_client is None:
+                        root_customer_client = customer_client
+                    # if it's level 0, we need to skip this round of loop
+                    continue
+
+                # For all level-1 (direct child) accounts that are a
```

Comment:
Nit: level 1, to match "level 0" above. Otherwise I read it as "level minus 1" 🤷‍♀️